### PR TITLE
BVAL-655 getContainerElementTypes() -> getConstrainedContainerElement…

### PR DESCRIPTION
**Pull request by gunnarmorling**
…Types(); adding example

https://hibernate.atlassian.net/browse/BVAL-655

### CODE REVIEW

File: sources/constraint-metadata.asciidoc [hunk 1]
```diff
@@ -386,7 +386,7 @@ include::{validation-api-source-dir}javax/validation/metadata/ReturnValueDescrip
 [[constraintmetadata-containerdescriptor]]
 === ContainerDescriptor and ContainerElementTypeDescriptor
 
-The `ContainerDescriptor` interface describes those elements that can host container element constraints (see <<constraintdeclarationvalidationprocess-containerelementconstraints>>).
+The `ContainerDescriptor` interface describes those elements that can be of a container type such `List` or `Map` and as such may host container element constraints (see <<constraintdeclarationvalidationprocess-containerelementconstraints>>) or have container element types that are marked with `@Valid`.
```
Comment (gsmet) [inline sources/constraint-metadata.asciidoc]: I think you're missing a `as` after the first `such`. But it's quite weird to have `as such` just after `such as`. Maybe replacing the first `such` with `e.g.` would be in order?

File: sources/constraint-metadata.asciidoc [hunk 2]
```diff
@@ -395,19 +395,160 @@ The `ContainerDescriptor` interface describes those elements that can host conta
...
 ----
 
-[tck-testable]#`getContainerElementTypes()` returns a list of descriptors representing all the type arguments of the container type,
-in the order of their declaration.#
+[tck-testable]#If a given element is of a container type, `getConstrainedContainerElementTypes()` returns a set with descriptors representing those container element types that either host at least one constraint or are marked with `@Valid`.#
+[tck-testable]#The returned set will be empty if the given element is not of a container type or is of a container type but has no element types that are constrained or marked with `@Valid`.#
+In the context of properties and method return values, container element types of super-types are considered.
 
-[tck-testable]#The `ContainerElementTypeDescriptor` interface describes the potential container element constraints applied to one type argument of a container type.#
+[tck-testable]#The `ContainerElementTypeDescriptor` interface describes the potential container element constraints applied to one element type a container.#
```
Comment (gsmet) [inline sources/constraint-metadata.asciidoc]: `element type a container` -> missing `of`

**Comment by gunnarmorling** [comment]
Reworked and force-pushed the commit to address your comments, @gsmet!

**Comment by gsmet** [comment]
Rebased and merged, thanks!

**Pull request closed without merging**